pkg/fluxApplication/FluxApplicationService.go

```diff
@@ -0,0 +1,126 @@
+}
+
+type FluxApplicationServiceImpl struct {
+	logger *zap.SugaredLogger
+	//helmAppService service.HelmAppService
+	clusterService cluster.ClusterService
+	helmAppClient  gRPC.HelmAppClient
+}
+
+func NewFluxApplicationServiceImpl(logger *zap.SugaredLogger,
+	//helmAppService service.HelmAppService,
+	clusterService cluster.ClusterService,
+	helmAppClient gRPC.HelmAppClient) *FluxApplicationServiceImpl {
+	return &FluxApplicationServiceImpl{
+		logger: logger,
+		//helmAppService: helmAppService,
+		clusterService: clusterService,
+		helmAppClient:  helmAppClient,
+	}
+
+}
+
+func (impl *FluxApplicationServiceImpl) ListApplications(ctx context.Context, clusterIds []int) ([]*bean.FluxApplicationListDto, error) {
+	var clusters []*cluster.ClusterBean
+	var err error
+	appListCluster := make([]*bean.FluxApplicationListDto, 0)
+	req := &gRPC.AppListRequest{}
+	if len(clusterIds) > 0 {
+		//for _, clusterId := range clusterIds {
+		//	clusterConfig, err := impl.helmAppService.GetClusterConf(clusterId)
+		//	if err != nil {
+		//		impl.logger.Errorw("error in getting clusters by ids", "err", err, "clusterIds", clusterIds)
+		//		return nil, err
+		//	}
+		//	req.Clusters = append(req.Clusters, clusterConfig)
+		//}
+
+	} else {
+		clusters, err = impl.clusterService.FindAll()
+		if err != nil {
+			impl.logger.Errorw("error in getting all active clusters", "err", err)
+			return nil, err
+		}
+
+		configs, err1 := impl.ConvertClusterBeanToClusterConfig(clusters)
+		if err1 != nil {
+			impl.logger.Errorw("error in getting all active clusters", "err", err1)
+			return nil, err1
+		}
+		req.Clusters = configs
+
+	}
+
+	//fluxAppsClusterCount := make(map[int32]int)
+
+	applicationStream, err := impl.helmAppClient.ListFluxApplication(ctx, req)
+	if err == nil {
+		fluxApplicationList, err1 := applicationStream.Recv()
+		if err1 != nil {
+			impl.logger.Errorw("error in list Flux applications streams recv", "err", err)
+		} else {
+			appLists := fluxApplicationList.GetFluxApplicationDetail()
+
+			for _, app := range appLists {
+				fluxAppListDto := make([]*bean.FluxApplicationDto, 0)
+				fluxAppListDto = append(fluxAppListDto, &bean.FluxApplicationDto{
+					Name:         app.Name,
+					SyncStatus:   app.SyncStatus,
+					HealthStatus: app.HealthStatus,
+					EnvironmentDetails: &bean.EnvironmentDetail{
+						Namespace:   app.EnvironmentDetail.Namespace,
+						ClusterId:   int(app.EnvironmentDetail.ClusterId),
+						ClusterName: app.EnvironmentDetail.ClusterName,
+					},
+				})
+
+				appListCluster = append(appListCluster, &bean.FluxApplicationListDto{
+					ClusterId:  int(fluxApplicationList.ClusterId),
+					FluxAppDto: fluxAppListDto,
+				})
+			}
+
+		}
+	} else {
+		impl.logger.Errorw("error while fetching list application from kubelink", "err", err)
+	}
+	return appListCluster, nil
+}
+
+func (impl *FluxApplicationServiceImpl) ConvertClusterBeanToClusterConfig(clusters []*cluster.ClusterBean) ([]*gRPC.ClusterConfig, error) {
```

Comment:
I think we already have this function defined somewhere, could you  please look for that and reuse that instead of creating a new one